wordcount_and_median.py

```diff
@@ -43,46 +42,32 @@ def __cleaner_lister(incoming_string):
     outgoing_list = incoming_string.split()
     return outgoing_list
 
-     
 
 def count_writer(output_folder, wc_out_file, fulldict_cw):
-    with open(output_folder+'/'+wc_out_file, 'w') as d:
-        for item in sorted( fulldict_cw.keys() ):
-            out_line = str(item)+'\t'+str( fulldict_cw[item] )+'\n'
-            d.write(out_line)
-        d.closed
+    with open(os.path.join(output_folder, wc_out_file), 'w') as d:
+        writer = csv.writer(d, delimiter='\t')
+        for word, count in sorted(fulldict_cw.items()):
+            writer.writerow([word, str(count)])
 
 
 def median_writer(output_folder, med_out_file, linelengths_mw):
-    sorted_list = [];
-    with open(output_folder+'/'+med_out_file, 'w') as m:
-        for item in range( len(linelengths_mw) ):
-            bisect.insort(sorted_list, linelengths_mw[item]);
-            m.write('{0:.1f}\n'.format( __presorted_median(sorted_list) ) )
-        m.closed
+    sorted_list = []
+    with open(os.path.join(output_folder, med_out_file), 'w') as m:
+        for linelength in linelengths_mw:
+            bisect.insort(sorted_list, linelength)
+            m.write('{0:.1f}\n'.format( _presorted_median(sorted_list) ) )
 
 
-def __presorted_median(data):
+def _presorted_median(data):
     n = len(data)
     if n == 0:
-        raise StatisticsError("no median for empty data")
-    if n%2 == 1:
-        return data[n//2]
+        raise statistics.StatisticsError('no median for empty data')
+    if n % 2 == 1:
+        return data[n // 2]
     else:
-        i = n//2
-        return (data[i - 1] + data[i])/2
-
-
-if __name__ == "__main__":
-    import sys
-    if len(sys.argv) == 3:
-        wc_m_main( str(sys.argv[1]), str(sys.argv[2]) )
-    elif len(sys.argv) == 2:
-        wc_m_main( str(sys.argv[1]) )
-    else:
-        wc_m_main()
- 
-
-
+        i = n // 2
+        return (data[i - 1] + data[i]) / 2
 
 
+if __name__ == '__main__':
+    wc_m_main(*sys.argv[1:])
```

Comment:
Basic overview of the ``*`` syntax:

```python
>>> def f(x, y): pass
... 
>>> 
>>> # calling a function normally
>>> f(5, 6)
>>> 
>>> # calling with keyword arguments
>>> f(x=5, y=6)
>>> f(y=6, x=5)
>>> 
>>> # calling with a combination of positional and keyword arguments
>>> f(5, y=6)
>>> 
>>> # sometimes we have a list and want to use that list as our arguments
>>> xs = [5, 6]
>>> 
>>> f(xs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: f() takes exactly 2 arguments (1 given)
>>> 
>>> # the above didn't work because we tried to pass the list itself as the first argument
>>> 
>>> # using the * syntax makes this work as we want
>>> f(*xs)
>>> 
>>> # you can even combine normal arguments and * arguments
>>> xs = [6]
>>> f(5, *xs)
```